reframe/utility/jsonext.py

```diff
@@ -18,10 +18,31 @@ def __rfm_json_encode__(self):
             '__rfm_file__': inspect.getfile(type(self))
         }
         ret.update(self.__dict__)
-        return ret
+        encoded_ret = encode_dict(ret, recursive=True)
+        _ret = encoded_ret if encoded_ret else ret
+        return _ret
 
 
-def encode(obj):
+def encode_dict(obj, *, recursive=False):
+    '''Transform tuple dict keys into strings.
+
+    Use the recursive option to also check the keys in nested dicts.
+    '''
+
+    if isinstance(obj, MutableMapping):
+        if recursive or any(isinstance(k, tuple) for k in obj):
+            newobj = type(obj)()
+            for k, v in obj.items():
+                _key = str(k) if isinstance(k, tuple) else k
+                _v = encode_dict(v, recursive=recursive)
+                newobj[_key] = _v if _v else v
```

Comment:
Better be explicit here about None:

```suggestion
                newobj[_key] = _v if _v is not None else v
```